### ISSUE TITLE
Vaadin-boot starts the app even if it fails to connect to the database

### ISSUE THREAD

**Issue by mvysny** [mvysny/jdbi-orm-vaadin-crud-demo#30]
When postgresql is unavailable, HikariCP will throw an exception in Bootstrap.contextInitialized(). However, the app still starts and blocks in Vaadin-boot. This makes Kubernetes think that the app started properly, even though it doesn't work afterwards.

**Comment by mvysny** [comment]
Patching `Main` to call `setThrowUnavailableOnStartupException(true);` on `WebAppContext` helps:
```java
public final class Main {
    public static void main(@NotNull String[] args) throws Exception {
        new VaadinBoot() {
            @Override
            protected @NotNull WebAppContext createWebAppContext() throws IOException {
                final WebAppContext ctx = super.createWebAppContext();
                ctx.setThrowUnavailableOnStartupException(true);
                return ctx;
            }
        }.withArgs(args).run();
    }
}
```
I think this should be the default in Vaadin-Boot.

**Comment by mvysny** [comment]
Filed https://github.com/mvysny/vaadin-boot/issues/14

**Comment by mvysny** [comment]
upstream fixed, i'll v-bump vaadin-boot version to 11.3 once the new version is released
